examples/external-virtual-network/main.tf

```diff
@@ -0,0 +1,86 @@
+terraform {
+  required_version = ">= 1.9, < 2.0"
+  required_providers {
+    azapi = {
+      source  = "Azure/azapi"
+      version = "~> 2.0"
+    }
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 4.0"
+    }
+    modtm = {
+      source  = "Azure/modtm"
+      version = "0.3.2"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "3.6.2"
+    }
+  }
+}
+
+provider "azurerm" {
+
+  features {
+    key_vault {
+      purge_soft_delete_on_destroy = false
+    }
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+    #     api_management {
+    # purge_soft_delete_on_destroy = false
+    #     }
+  }
+}
+
+
+## Section to provide a random Azure region for the resource group
+# This allows us to randomize the region for the resource group.
+module "regions" {
+  source  = "Azure/regions/azurerm"
+  version = "~> 0.3"
+}
+
+# This allows us to randomize the region for the resource group.
+resource "random_integer" "region_index" {
+  max = length(module.regions.regions) - 1
+  min = 0
+}
+## End of section to provide a random Azure region for the resource group
+
+# This ensures we have unique CAF compliant names for our resources.
+module "naming" {
+  source  = "Azure/naming/azurerm"
+  version = "~> 0.3"
+}
+
+# This is required for resource modules
+resource "azurerm_resource_group" "this" {
+  location = module.regions.regions[random_integer.region_index.result].name
```

Comment:
I had to change this to "eastus2" to get this sample to work. 


I get the following error before the substitution
```
│ Service Name: "apimwufd"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: InvalidParameters: Invalid parameter: Location of SubnetResource :'eastus2' does not match the Location in Config:'westus2'
│
│   with module.test.azurerm_api_management.this,
│   on ..\..\main.tf line 1, in resource "azurerm_api_management" "this":
│    1: resource "azurerm_api_management" "this" {
```

Reply:
figured out the problem, in the examples `internal_virtual_network` and `external_virtual_network` I hardcoded the subnet id. I removed that hard coded and required the users to provide a subnet_id in the variables. I also have 2 examples where I create a preprovisioned vnet and associated resource.